### ISSUE TITLE
[openqa-mon][improvement] Possible optimization on API calls

### ISSUE THREAD

**Issue by ilmanzo** [os-autoinst/openqa-mon#159]
Once we have pagination in ( #157 ), would be nice to optimize by requesting a status refresh from the remotes only for the jobs actually visible, avoiding to fetch data for something that won't even be displayed. 
Need to coordinate/update in case of terminal resize.

**Comment by grisu48** [comment]
This doesn't work well with the notification feature, where we need to check the status of all actively monitored jobs.

**Comment by ilmanzo** [comment]
good point; one can want to be notified even for a non-displayed job (I guess that's the point of notifications) . Maybe it requires a decoupling of the information actually fetched from the server, and what is actually displayed , or request the user to "subscribe" with a keyboard shortcut to those jobs who wants get notified.

**Comment by grisu48** [comment]
While that's certainly possible, I don't know if it's worth implementing, as it would make the fetch logic more complex with little to no benefits. What do you think?

**Comment by ilmanzo** [comment]
I agree that's a big change, might not worth but it would be a nice to have, perhaps we can add something as a feature wishlist for next major (or GUI) version ?

OTOH, if the user has no notifications enabled (explicit config or used `--no-bell`                     
  `--no-notify` )  this kind of optimization can be put in place
